LICENSE

```diff
@@ -34,3 +26,7 @@ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
 CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 THE POSSIBILITY OF SUCH DAMAGE.
+
+   END OF TERMS AND CONDITIONS
+
+The generated code can be used under CC0 (public domain) license: https://creativecommons.org/share-your-work/public-domain/cc0/
```

Comment:
Well, this is not actually correct. The generated code can be used under any terms the end user wants. They can release it to the public domain if they want, or they can have it be closed source completely. But, generally, I don't understand why you would feel a desire to delete my previous statement, that any code you generate with the tool is *completely unencumbered*. Granted, I am not a lawyer myself, so obviously, if there was a real lawyer involved who could state the things with the absolutely correct legal terminology, but really, I think this is a very clear statement of the situation. The code you generate with the tool is just yours. It's not in the public domain! Unless you want it to be.

Now, of course, if you think that the LICENSE text should only conclude the actual BSD license text with no clarification, then the statement about the generated code being yours, completely unencumbered, could be moved to the README. That would be okay too, but I am very far from convinced there is much of a real problem with all this as things stand.

Reply:
@revusky , I see your pain.

However, as you say, you are not a lawyer.
You don't understand that, and it is OK.

TL;DR: `CC0` is a very-well studied and understood form of saying "do whatever you want license" in such a way that lawyers are happy.

---


However, when you invent custom rules (e.g. `completely unencumbered`), you basically mean the following: "everybody who considers using JavaCC must ask their lawyer to carefully study the license text and evaluate if the terms are acceptable or not".

There's no other way. Every single person who considers using JavaCC must ask their lawyers. Is it something you want to achieve?

Note: your `completely unencumbered` can't be an excuse to slip the lawyer.

----

Here's my suggestion: the generated code should be licensed under such a license that everybody understands (e.g. CC0).

For instance, if the text says that "the generated code can be used under CC0 terms", then the user **can** skip lawyer. In 99% of the cases, lawyers have already studied CC0 text, and it is extremely likely that CC0 is approved.

For instance, Apache Software Foundation has the list of **resolved** cases: https://www.apache.org/legal/resolved.html#handling-public-domain-licensed-works

It does list CC0 as an approved license.

As you can see, the list says it is forbidden to use the software that is licensed under [The "Don't Be A Dick" Public License](https://www.apache.org/legal/resolved.html#category-x)

However, the list says nothing on `completely unencumbered`. So, technically speaking, if somebody tries to use JavaCC for an Apache project, they must raise a question to lawyers.

Reply:
@revusky , I see your pain.

However, as you say, you are not a lawyer.
You don't understand that, and it is OK.

TL;DR: `CC0` is a very-well studied and understood form of saying "do whatever you want license" in such a way that lawyers are happy.

---


However, when you invent custom rules (e.g. `completely unencumbered`), you basically mean the following: "everybody who considers using JavaCC must ask their lawyer to carefully study the license text and evaluate if the terms are acceptable or not".

There's no other way. Every single person who considers using JavaCC must ask their lawyers. Is it something you want to achieve?

Note: your `completely unencumbered` can't be an excuse to slip the lawyer.

----

Here's my suggestion: the generated code should be licensed under such a license that everybody understands (e.g. CC0).

For instance, if the text says that "the generated code can be used under CC0 terms", then the user **can** skip lawyer. In 99% of the cases, lawyers have already studied CC0 text, and it is extremely likely that CC0 is approved.

For instance, Apache Software Foundation has the list of **resolved** cases: https://www.apache.org/legal/resolved.html#handling-public-domain-licensed-works

It does list CC0 as an approved license.

As you can see, the list says it is forbidden to use the software that is licensed under [The "Don't Be A Dick" Public License](https://www.apache.org/legal/resolved.html#category-x)

However, the list says nothing on `completely unencumbered`. So, technically speaking, if somebody tries to use JavaCC for an Apache project, they must raise a question to lawyers.